client/card-readers/preview-receipt/preview-receipt.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useState, useEffect } from '@wordpress/element';
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+/**
+ * Internal dependencies
+ */
+import { LoadableBlock } from '../../components/loadable';
+import PrintedReceiptPreviewer from 'wcpay/card-readers/preview-receipt/printed-receipt-previewer';
+import {
+	useAccountBusinessSupportAddress,
+	useAccountBusinessName,
+	useAccountBusinessURL,
+	useAccountBusinessSupportEmail,
+	useAccountBusinessSupportPhone,
+} from '../../data';
+import { FetchReceiptPayload } from 'wcpay/types/card-readers';
+
+async function fetchReceiptHtml(
+	payload: FetchReceiptPayload
+): Promise< string > {
+	const path = '/wc/v3/payments/readers/receipts/print/preview';
+	return apiFetch( { path, data: payload, method: 'post' } );
+}
+
+const PreviewReceipt = (): JSX.Element => {
+	const [ receiptHtml, setReceiptHtml ] = useState< string >( '' );
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ isErrorFetchingReceipt, setIsErrorFetchingReceipt ] = useState(
+		false
+	);
+
+	const [
+		accountBusinessSupportAddress,
+	] = useAccountBusinessSupportAddress();
+	const [ accountBusinessName ] = useAccountBusinessName();
+	const [ accountBusinessURL ] = useAccountBusinessURL();
+	const [ accountBusinessSupportEmail ] = useAccountBusinessSupportEmail();
+	const [ accountBusinessSupportPhone ] = useAccountBusinessSupportPhone();
+
+	useEffect( () => {
+		let didCancel = false;
+		async function fetchReceiptHtmlAPI() {
+			try {
+				const data = await fetchReceiptHtml( {
+					accountBusinessName,
+					accountBusinessSupportAddress,
+					accountBusinessURL,
+					accountBusinessSupportEmail,
+					accountBusinessSupportPhone,
+				} as FetchReceiptPayload );
+
+				if ( ! didCancel && data ) {
+					setIsLoading( false );
+					setReceiptHtml( data );
+				}
+			} catch ( error ) {
+				setIsLoading( false );
+				setIsErrorFetchingReceipt( true );
+			}
+		}
+		fetchReceiptHtmlAPI();
```

Comment:
Is `await`ing here not desirable?

Reply:
Thanks for checking it but I don't think so :) The inner function `fetchReceiptHtmlAPI` is marked `async` because it does `await` for `fetchReceiptHtml` in the body of this function. But in the `useEffect` callback, `fetchReceiptHtmlAPI` gets called, but it is not awaited for, if this makes sense.
In the body of `fetchReceiptHtmlAPI`, once the `fetchReceiptHtml` resolves (after the await), the state is populated calling the corresponding hooks: `setIsLoading` and `setReceiptHtml`.
This is my understanding but please let me know if I am missing something.

Reply:
In my opinion `await` is not required in the above case.